documentation/test/test_process_provider.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+import local_gen
+
+from unittest.mock import mock_open, patch, MagicMock
+from pathlib import Path
+
+class TestProcessProvider:
+    @pytest.fixture
+    def test_path(self):
+        return Path("/")
+    
+    @pytest.fixture
+    def test_string(self):
+        return "aws"
+
+    def test_provider_is_not_directory(self, test_path):
+        with patch.object(Path, 'is_dir', return_value=False) as mock_is_dir:
+            result = local_gen.process_provider(test_path, test_path, test_path, test_path, 0, [], {})
+            assert result == 0
+            mock_is_dir.assert_called_once()
+    
+    def test_metadata_failure(self, test_path):
+        mock_metadata_data = """{ 
+        no_data
+        }""" # Not parsable data
+        m = mock_open(read_data=mock_metadata_data)
+        
+        with patch.object(Path, 'is_dir', side_effect=[True, False]) as mock_is_dir:
+            with patch.object(Path, 'iterdir', return_value=[Path]):
+                with patch.object(Path, 'exists', return_value=True):
+                    with patch('builtins.open', m):
+                        with pytest.raises(Exception):
+                            local_gen.process_provider(test_path, test_path, test_path, test_path, 0, [], {})
+                            mock_is_dir.assert_called()
```

Comment:
suggestion: the [with](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement) statement has support for multiple context manager. The following syntax is semantically equivalent to what you wrote but arguably easier to read:
```suggestion
        with (
            patch.object(Path, 'is_dir', side_effect=[True, False]) as mock_is_dir,
            patch.object(Path, 'iterdir', return_value=[Path]),
            patch.object(Path, 'exists', return_value=True),
            patch('builtins.open', m),
            pytest.raises(Exception),
        ):
            local_gen.process_provider(test_path, test_path, test_path, test_path, 0, [], {})
            mock_is_dir.assert_called()
```